src/webAppCreator.ts

```diff
@@ -0,0 +1,484 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { AzureAccountWrapper } from './azureAccountWrapper';
+import { WizardBase, WizardResult, WizardStep, UserCancelledError } from './wizard';
+import { SubscriptionModels, ResourceManagementClient, ResourceModels } from 'azure-arm-resource';
+import WebSiteManagementClient = require('azure-arm-website');
+import * as WebSiteModels from '../node_modules/azure-arm-website/lib/models';
+import * as util from './util';
+
+export class WebAppCreator extends WizardBase {
+    constructor(output: vscode.OutputChannel, readonly azureAccount: AzureAccountWrapper) {
+        super(output);
+        this.steps.push(new SubscriptionStep(this, azureAccount));
+        this.steps.push(new ResourceGroupStep(this, azureAccount));
+        this.steps.push(new AppServicePlanStep(this, azureAccount));
+        this.steps.push(new WebsiteStep(this, azureAccount));
+    }
+
+    protected beforeExecute(step: WizardStep, stepIndex: number) {
+        if (stepIndex == 0) {
+            this.writeline('Start creating new Web App...');
+        }
+    }
+
+    protected onExecuteError(step: WizardStep, stepIndex: number, error: Error) {
+        this.writeline(`Failed to create new Web App - ${error.message}`);
```

Comment:
If the user cancelled, then it's not really a 'Failure' and this shouldn't be displayed